Classes/TranslationAdapter/OpenAiPhpTranslationAdapter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TYPO3Headless\Typo3Ai\TranslationAdapter;
+
+/*
+ * This file is part of the Macopedia. package.
+ *
+ * (c) 2023 Macopedia <extensions@macopedia.pl>, macopedia.com
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use OpenAI\Client;
+use TYPO3Headless\Typo3Ai\Credentials\CredentialsInterface;
+use TYPO3Headless\Typo3Ai\ModelType\ModelTypeInterface;
+
+class OpenAiPhpTranslationAdapter implements TranslationAdapterInterface
+{
+    public const NO_TRANSLATION = 'NO_TRANSLATION';
+    public const TYPE = 'openai-php/client';
+
+    protected bool $initialized = false;
+
+    protected ModelTypeInterface|null $type = null;
+
+    protected ?Client $client = null;
+
+    protected array $defaultConfig = [
+        'model' => '',
+        'messages' => [],
+    ];
+
+    protected string $prompt = 'Translate text (keep html unchanged) after next semi-colon to %s %s or write \'' . self::NO_TRANSLATION . '\'; ';
```

Comment:
I think that we should be able to set prompt via configuration settings, that's good for now, but keep this in mind